core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java

```diff
@@ -16,52 +16,51 @@
 import de.jplag.reporting.FilePathUtil;
 import de.jplag.reporting.reportobject.model.ComparisonReport;
 import de.jplag.reporting.reportobject.model.Match;
-import de.jplag.reporting.reportobject.writer.JsonWriter;
+import de.jplag.reporting.reportobject.writer.JPlagResultWriter;
 
 /**
  * Writes {@link ComparisonReport}s of given {@link JPlagResult} to the disk under the specified path. Instantiated with
  * a function that associates a submission to its id.
  */
 public class ComparisonReportWriter {
 
-    private final JsonWriter fileWriter;
+    private final JPlagResultWriter resultWriter;
     private final Function<Submission, String> submissionToIdFunction;
     private final Map<String, Map<String, String>> submissionIdToComparisonFileName = new ConcurrentHashMap<>();
     private final Map<String, AtomicInteger> fileNameCollisions = new ConcurrentHashMap<>();
 
-    public ComparisonReportWriter(Function<Submission, String> submissionToIdFunction, JsonWriter fileWriter) {
+    public ComparisonReportWriter(Function<Submission, String> submissionToIdFunction, JPlagResultWriter jPlagResultWriter) {
         this.submissionToIdFunction = submissionToIdFunction;
-        this.fileWriter = fileWriter;
+        this.resultWriter = jPlagResultWriter;
     }
 
     /**
      * Generates detailed ComparisonReport DTO for each comparison in a JPlagResult and writes them to the disk as json
      * files.
      * @param jPlagResult The JPlagResult to generate the comparison reports from. contains information about a comparison
-     * @param path The path to write the comparison files to
      * @return Nested map that associates each pair of submissions (by their ids) to their comparison file name. The
      * comparison file name for submission with id id1 and id2 can be fetched by executing get two times:
      * map.get(id1).get(id2). The nested map is symmetrical therefore, both map.get(id1).get(id2) and map.get(id2).get(id1)
      * yield the same result.
      */
-    public Map<String, Map<String, String>> writeComparisonReports(JPlagResult jPlagResult, String path) {
+    public Map<String, Map<String, String>> writeComparisonReports(JPlagResult jPlagResult) {
         int numberOfComparisons = jPlagResult.getOptions().maximumNumberOfComparisons();
         List<JPlagComparison> comparisons = jPlagResult.getComparisons(numberOfComparisons);
-        writeComparisons(path, comparisons);
+        writeComparisons(comparisons);
         return submissionIdToComparisonFileName;
     }
 
-    private void writeComparisons(String path, List<JPlagComparison> comparisons) {
-        comparisons.parallelStream().forEach(comparison -> {
+    private void writeComparisons(List<JPlagComparison> comparisons) {
+        for (JPlagComparison comparison : comparisons) {
```

Comment:
I would suggest also performing a test with a lot of very small submissions (1000 maybe?, but each only a few small files), to see whether removing the parralelism actually decreased performance here